docs/extensions/server/applications.rst

```diff
@@ -20,5 +20,117 @@ Manage Applications against Extra Cluster Managers
 
 .. caution:: unstable
 
-.. warning::
-   This page is still in-progress.
+Kyuubi supports configuring custom ``org.apache.kyuubi.engine.ApplicationOperation`` for extra cluster manager which provides an ability to control application, including getting information, killing.
+
+   .. code-block:: scala
+
+      trait ApplicationOperation {
+
+        /**
+         * Step for initializing the instance.
+         */
+        def initialize(conf: KyuubiConf): Unit
+
+        /**
+         * Step to clean up the instance
+         */
+        def stop(): Unit
+
+        /**
+         * Called before other method to do a quick skip
+         *
+         * @param clusterManager the underlying cluster manager or just local instance
+         */
+        def isSupported(clusterManager: Option[String]): Boolean
+
+        /**
+         * Kill the app/engine by the unique application tag
+         *
+         * @param tag the unique application tag for engine instance.
+         *            For example,
+         *            if the Hadoop Yarn is used, for spark applications,
+         *            the tag will be preset via spark.yarn.tags
+         * @return a message contains response describing how the kill process.
+         *
+         * @note For implementations, please suppress exceptions and always return KillResponse
+         */
+        def killApplicationByTag(tag: String): KillResponse
+
+        /**
+         * Get the engine/application status by the unique application tag
+         *
+         * @param tag the unique application tag for engine instance.
+         * @return [[ApplicationInfo]]
+         */
+        def getApplicationInfoByTag(tag: String): ApplicationInfo
+      }
+
+      /**
+        * (killed or not, hint message)
+        */
+      type KillResponse = (Boolean, String)
+
```

Comment:
.. code-block:: scala

and describe it a bit, see an example I give it to you https://github.com/apache/incubator-kyuubi/pull/3178/files#r936582755

Reply:
updated